flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -813,7 +813,7 @@ private ApplicationReport startAppMaster(
 
 		// Setup jar for ApplicationMaster
 		Path remotePathJar = setupSingleLocalResource(
-				"flink.jar",
+				flinkJarPath.getName(),
```

Comment:
This has potential for failures if the name of the local jar contains characters that aren't supported by the target filesystem.

Reply:
We call `Path.getName` where `Path` is `org.apache.hadoop.fs.Path` so I think we don't fail into this case. BTW, we already ship user jars in this pattern. So if this is valid, we also suffer from the risk that user jars contains characters that aren't supported by the target filesystem.